### PR TITLE
Fixed datepicker issue

### PR DESCRIPTION
**Pull request by 20BBrown14**
Fixed the datepicker issue that we were seeing in consumer apps when trying to render it. The error had a message like `MUI Utils not in context`.

This PR effectively removes the need to use the `--legacy-peer-deps` flag when installing node modules. This required updating a bunch of packages. I haven't found any issues as a result of these upgrades which I tested in an ancillary app and in sc2-senior.

I'm still not 100% what was causing the issue but mismatch MUI dep versions, or a broken dep tree due to the the `--legacy-peer-dep` flag could have been the culprit.

This should release v10.1.1

Closes: #832

### CODE REVIEW

**Review by laurelbean — APPROVED** [review]
Thanks for taking this on!

File: src/components/SQForm/SQFormReadOnlyField.tsx [hunk 1]
```diff
@@ -42,7 +42,6 @@ function SQFormReadOnlyField({
         InputProps={{
           ...InputProps,
           readOnly: true,
-          disableUnderline: true,
```
Comment (hvilander) [inline src/components/SQForm/SQFormReadOnlyField.tsx:45]: I assume this is a new default?

File: package.json [hunk 2]
```diff
@@ -62,12 +62,10 @@
   "lint-staged": {
     "*.{js,ts,tsx}": [
       "prettier --write",
-      "eslint --fix",
-      "git add"
```
[the remark this answers is not in the document]
Comment (20BBrown14) [reply package.json:66]: The latest version of the `lint-staged` package automatically does a `git add` after each command so no need to have it in our command list

File: src/components/SQForm/SQFormReadOnlyField.tsx (hunk 1 above)
Comment (20BBrown14) [reply src/components/SQForm/SQFormReadOnlyField.tsx:45]: I believe this prop is no longer a prop after upgrading. Which is why it was removed.

**Review by hvilander — APPROVED** [review]

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (hvilander) [reply package.json:66]: :party parrot:

File: src/components/SQForm/SQFormReadOnlyField.tsx (hunk 1 above)
Comment (hvilander) [reply src/components/SQForm/SQFormReadOnlyField.tsx:45]: fair

**Review by hvilander — APPROVED** [review]

**Review by samasastudio — APPROVED** [review]

**Review by aharpt — APPROVED** [review]

**Pull request merged**